Rewrite HTTP OAuth2 client BBEs (#3793)
ballerina-platform/ballerina-distribution: examples/http-client-oauth2-password-grant-type/http_client_oauth2_password_grant_type.md

# HTTP client - OAuth2 password grant type

> The `http:Client` can be secured with OAuth2 password grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this grant type when you need to exchange the user's credentials for an access token.

Author: ```suggestion
The `http:Client` can connect to a service that is secured with the OAuth2 password grant type by adding the `Authorization: Bearer <token>` header to each request. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this grant type when you need to exchange the user's credentials for an access token.
```

> An `http:Client` can be secured with OAuth2 password grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 password grant type can be specified in the `auth` field as the `http:OAuth2PasswordGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.

Reviewer: **When to use ?**
use when you need to exchange a user's credentials for an access token. Use this only when there is high level of trust with the client. This grant type is not recommended as this grant type is deprecated per OAuth 2.1.

Author: Added via [187d99d](https://github.com/ballerina-platform/ballerina-distribution/pull/3793/commits/187d99dffcd52f92e019e05f224ac4dc5e4a2e66)

Author: I did not add the deprecated statement to the content. If this is deprecated we may remove this BBE right?

> An `http:Client` can be secured with OAuth2 password grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 password grant type can be specified in the `auth` field as the `http:OAuth2PasswordGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.

Reviewer 2: ```suggestion
The `http:Client` can be secured with OAuth2 password grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 password grant type can be specified in the `auth` field as the `http:OAuth2PasswordGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.
```

> The `http:Client` can be secured with OAuth2 password grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 password grant type can be specified in the `auth` field as the `http:OAuth2PasswordGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.

Author: ```suggestion
The `http:Client` can be secured with OAuth2 password grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 password grant type can be specified in the `auth` field as the `http:OAuth2PasswordGrantConfig` record. Use this grant type when you need to exchange the user's credentials for an access token.
```

> The `http:Client` can be secured with OAuth2 password grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 password grant type can be specified in the `auth` field as the `http:OAuth2PasswordGrantConfig` record. Use this grant type when you need to exchange the user's credentials for an access token.

Author: ```suggestion
The `http:Client` can be secured with OAuth2 password grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this grant type when you need to exchange the user's credentials for an access token.
```
